test/reviewController.test.js

```diff
@@ -0,0 +1,113 @@
+const { app, connectDB } = require('../src/expressServer'); // Import your Express app here
+const { StatusCodes } = require('http-status-codes');
+const request = require('supertest');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const Review = require('../src/models/Review');
+const User = require('../src/models/User');
+const Album = require('../src/models/Album');
+
+let server;
+let mongooseConnection;
+let mongodb;
+
+beforeAll(async () => {
+  mongodb = await MongoMemoryServer.create();
+  const url = mongodb.getUri();
+  process.env.MONGO_URL = url;
+  mongooseConnection = await connectDB(url);
+  server = await app.listen(8001);
+});
+
+afterAll(async () => {
+  await server.close();
+  await mongooseConnection.disconnect();
+  await mongodb.stop();
+});
+
+describe('ReviewController API Tests', () => {
+  beforeAll(async () => {
+    await Review.deleteMany({});
+    await User.create({
+      email: 'Emily@google.com',
+      password: 'secret',
+      name: 'Emily',
+      username: 'emily123',
+      role: 'user',
+    });
+  });
+
+  beforeEach(async () => {
+    await Review.deleteMany({});
+    // Create a unique album and user for each test
+    const album = await Album.create({
+      albumName: 'Unique Album',
+      artistName: 'Unique Artist',
+      spotifyUrl: 'https://api.spotify.com/v1/albums/unique',
+    });
+
+    const user = await User.findOne({ username: 'emily123' });
+
+    const mockReviewData = {
+      rating: 5,
+      title: 'Loved this review',
+      comment: 'Recommend it"',
+      user: user._id,
+      album: album._id,
+    };
+
+    await Review.create(mockReviewData);
+  });
+  //get all reviews
+
+  it('should test the getAllReviews endpoint - Success Case', async () => {
+    const response = await request(app).get('/api/v1/reviews');
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('reviews');
```

Comment:
could we also check for the presence and correct value of the `count` property of the response body in this test?

Reply:
done